packages/core/js-client/src/keypair/index.ts

```diff
@@ -53,7 +57,13 @@ export class KeyPair {
     static async randomEd25519(): Promise<KeyPair> {
         const key = await generateKeyPair('Ed25519');
         const lib2p2Pid = await createFromPrivKey(key);
-        return new KeyPair(key, lib2p2Pid);
+        return new KeyPair(key, key.public, lib2p2Pid);
+    }
+    
+    static async fromPublicKey(publicKey: Uint8Array): Promise<KeyPair> {
```

Comment:
KeyPair cannot be created from PublicKey only. Key _Pair_ is a pair of keys: private and public. It can only be created from private key.

I guess what you want here instead, is `VerifyingKey(pk: PublicKey)`, or just plainly use libp2p's `PublicKey` directly. You're gonna use it in a single place, so that's fine.